server/src/main/java/io/crate/execution/ddl/TransportSchemaUpdateAction.java

```diff
@@ -302,16 +302,19 @@ private static void populateColumnPositions(Map<String, Object> mapping,
             contentPath.add(name);
             Map<String, Object> columnProperties = (Map<String, Object>) e.getValue();
             columnProperties = furtherColumnProperties(columnProperties);
+            assert columnProperties.containsKey("position") && columnProperties.get("position") != null : "Column position is missing: " + name;
+            // BWC compatibility with nodes < 5.1, position could be NULL if column is created on that nodes
             Integer position = (Integer) columnProperties.get("position");
-            assert position != null : "position should not be null";
-            if (position < 0) {
-                columnPositionResolver.addColumnToReposition(contentPath.pathAsText(""),
-                                                             position,
-                                                             columnProperties,
-                                                             (cp, p) -> cp.put("position", p),
-                                                             contentPath.currentDepth());
-            } else {
-                maxColumnPosition[0] = Math.max(maxColumnPosition[0], position);
+            if (position != null) {
```

Comment:
Does it make sense to continue on possible child columns if the position of the current one is not set?
I think we can just `continue` on null position values and avoid any further processing.

Reply:
Currently, we map `null` to `0` in `DocIndexMetadata` and this is affecting `position` fields in `Mapper` instances (mapper instances with position 0). In few places, we do
``` 
// ObjectMapper#toXcontent()
if (position != NOT_TO_BE_POSITIONED) { // if (position != 0) // this is here to filter MetadataFieldMappers and RootObjectMapper
            builder.field("position", position);
        }
```

So, 
```
null
  └ -1
```
this could be a possible case.

Reply:
It would be great if I could get some guidance on whether `Mapper.position` or `Mapper.Builder.position` could stay as typed `int` or reverted back to `Integer` to be able to contain `null`.

Reply:
Fixed, Thank you

Reply:
> Currently, we map null to 0 in DocIndexMetadata and this is affecting position fields in Mapper instances (mapper instances with position 0).

I don't really understand your described relation between `DocIndexMetadata` and the `Mapper` instances, `DocIndexMetadata` is only reading the metadata and never writes anything. 
Anyhow, as my suggestion would be more an optimization, I'm also fine to merge it as it.